### PR TITLE
623 datepickers

### LINKED ISSUE #623: Datepicker

### PR DESCRIPTION
**Pull request by kronken**
Closes #623 

Date picker wont close by itself, which might be a problem.

### CODE REVIEW

File: src/components/sidebar/routePathView/routePathInfoTab/RoutePathForm.tsx [hunk 1]
```diff
@@ -125,15 +125,19 @@ class RoutePathForm extends React.Component<IRoutePathFormProps>{
                 <div className={s.flexRow}>
                     <InputContainer
                         label='VOIM. AST'
-                        value={Moment(routePath.startTime)
-                            .format(datetimeStringDisplayFormat)}
+                        type='date'
+                        value={routePath.startTime}
+                        onChange={this.onChange('startTime')}
                         disabled={isEditingDisabled}
+                        validatorRule={routePathValidationModel.date}
```
Comment (Divinare) [inline src/components/sidebar/routePathView/routePathInfoTab/RoutePathForm.tsx]: Should we have routePathValidationModel validation fields with the same name as in IRoutePath?
routePathValidationModel.startTime
routePathValidationModel.endTime

File: src/components/sidebar/InputContainer.tsx [hunk 2]
```diff
@@ -65,6 +68,9 @@ class InputContainer extends React.Component<IInputProps, IInputState> {
     }
 
     render() {
+        const type = this.props.type || 'text';
+        const dateStringDisplayFormat = 'DD.MM.YYYY';
```
Comment (Divinare) [inline src/components/sidebar/InputContainer.tsx]: I wouldn't put this into a helper variable, just use 'DD.MM.YYYY' straightly

File: src/components/sidebar/InputContainer.tsx [hunk 3]
```diff
@@ -80,22 +86,31 @@ class InputContainer extends React.Component<IInputProps, IInputState> {
...
+                                classnames(
+                                    this.props.className,
+                                    this.props.disabled ? s.disabled : null,
+                                    !this.state.isValid ? s.invalidInput : null)
+                            }
+                            disabled={this.props.disabled}
+                            value={this.props.value ? (this.props.value as string | number) : ''}
+                            onChange={this.onChange}
+                        />
+                    )
```
[the remark this answers is not in the document]
Comment (kronken) [reply src/components/sidebar/InputContainer.tsx]: Did it a bit differently, but also in a much nicer fashion than before.

File: src/components/sidebar/InputContainer.tsx [hunk 4]
```diff
@@ -14,7 +16,7 @@ interface IInputProps {
     value?: string|number|Date;
     validatorRule?: string;
     icon?: React.ReactNode;
-    type?: 'text' | 'number' | 'date';
+    type?: inputType;
```
Comment (Divinare) [inline src/components/sidebar/InputContainer.tsx]: What if we make type as a required property?

File: src/components/sidebar/InputContainer.tsx [hunk 5]
```diff
@@ -81,26 +123,10 @@ class InputContainer extends React.Component<IInputProps, IInputState> {
                     </div>
                     }
                 </div>
```
Comment (Divinare) [inline src/components/sidebar/InputContainer.tsx]: Move this to renderInputLabel()?

```
                <div className={s.inputLabel}>
                    {this.props.label}
                    {!this.props.disabled && this.props.icon && this.props.onIconClick &&
                    <div
                        className={classnames(s.inline, s.pointer)}
                        onClick={this.props.onIconClick!}
                    >
                        {this.props.icon}
                    </div>
                    }
                </div>
```

File: src/components/sidebar/InputContainer.tsx (hunk 4 above)
Comment (kronken) [reply src/components/sidebar/InputContainer.tsx]: I've tried to simulate the HTML5 input element, where type also is optional. Empty type field will in both this, and HTML 5 result in a text input.

**Pull request merged**